### PR TITLE
Parametrize all IO tests

### PR DESCRIPTION
**Pull request by phillord**
I have started work on paramertisizing all the roundtrip tests (which should be all the tests in XML writer, RDF reader and writer and functional when it arrives).

As expected, I have found one bug already from a missing test.

### CODE REVIEW

**Comment by phillord** [comment]
Currently blocked on https://github.com/phillord/pretty_rdf/issues/6

**Comment by phillord** [comment]
All ready to go. Any new file added to the bubo directory now will be tested for XML round tripping, comparison between XML and RDF, and RDF roundtripping. XML reading needs to be testing explicitly (although we also read XML for many of the other tests) and is where the majority of hand-written checks should go.

For functional and any other format that we choose, we can just replicate the XML -> XXX and XXX roundtripping comparison.

File: Cargo.toml [hunk 1]
```diff
@@ -29,10 +29,12 @@ ureq={version="2.1.1", optional=true}
 
 [workspace]
 members=["horned-bin"]
+default-members=[".", "horned-bin"]
 
 [workspace.dependencies]
 indexmap="1.0.2"
-pretty_rdf="0.2.0"
+##pretty_rdf={path="./pretty_rdf"}
```
Comment (filippodebortoli) [inline Cargo.toml:36]: Could this line be eliminated?

File: src/io/rdf/writer.rs [hunk 2]
```diff
@@ -33,16 +33,27 @@ pub fn write<A: ForIRI, AA: ForIndex<A>, W: Write>(
...
+        let mut f = PrettyRdfXmlFormatter::new(write, ChunkedRdfXmlFormatterConfig::all().prefix(p))?;
+        ont.render(&mut f, &mut bng)?;
+        f.finish()?;
+    }
+
+
 
     // for i in f.triples() {
     //     println!("{}", i.printable());
     // }
```
Comment (filippodebortoli) [inline src/io/rdf/writer.rs:55]: Will we ever need this debugging loop again, or can we delete it?

File: horned-bin/src/bin/horned_triples.rs [hunk 3]
```diff
@@ -107,9 +109,9 @@ pub(crate) fn matcher(matches: &ArgMatches) -> Result<(), HornedError> {
 
         let mut f: pretty_rdf::PrettyRdfXmlFormatter<String, _> =
             pretty_rdf::PrettyRdfXmlFormatter::new(
-                b,
-                pretty_rdf::ChunkedRdfXmlFormatterConfig::all(),
-            )?;
+                 b,
+                 pretty_rdf::ChunkedRdfXmlFormatterConfig::all(),
+             )?;
         //let mut f = rio_xml::RdfXmlFormatter::with_indentation(&b, 4)?;
```
Comment (filippodebortoli) [inline horned-bin/src/bin/horned_triples.rs:115]: Can this line be removed, or does it still serve a purpose for future work?

**Review by filippodebortoli — APPROVED** [review]
Approved. I pointed out few sections of commented code that looked like debugging leftovers to be removed.

**Comment by phillord** [comment]
Yes, I will merge. I am inclined to leave all of those debugging lines in, as well as this which is even worse:

https://github.com/phillord/horned-owl/blob/9c1048bc12618cb46015befaa6ee8295d5f83fc8/src/io/rdf/writer.rs#L40-L49

At same point the writer should be configurable. It should also be able to produce turtle, ntriples and the like via rio (or a rio wrapper).

**Comment by althonos** [comment]
@phillord : There is a chance I'll just close the PR and start from a fresh rebase, I can foresee the merge errors on every test file happening and that's gonna be quicker to just rewrite the parser code in the current library :+1:

**Pull request merged**